algorithms/src/Kokkos_NestedSort.hpp

```diff
@@ -14,175 +14,17 @@
-                                             const ValueViewType& valueView) {
-  Impl::sort_nested_impl(t, keyView, valueView,
-                         Experimental::Impl::StdAlgoLessThanBinaryPredicate<
-                             typename KeyViewType::non_const_value_type>(),
-                         Impl::NestedRange<true>());
-}
-
-template <class TeamMember, class KeyViewType, class ValueViewType,
-          class Comparator>
-KOKKOS_INLINE_FUNCTION void sort_by_key_team(const TeamMember& t,
-                                             const KeyViewType& keyView,
-                                             const ValueViewType& valueView,
-                                             const Comparator& comp) {
-  Impl::sort_nested_impl(t, keyView, valueView, comp,
-                         Impl::NestedRange<true>());
-}
-
-template <class TeamMember, class ViewType>
-KOKKOS_INLINE_FUNCTION void sort_thread(const TeamMember& t,
-                                        const ViewType& view) {
-  Impl::sort_nested_impl(t, view, nullptr,
-                         Experimental::Impl::StdAlgoLessThanBinaryPredicate<
-                             typename ViewType::non_const_value_type>(),
-                         Impl::NestedRange<false>());
-}
-
-template <class TeamMember, class ViewType, class Comparator>
-KOKKOS_INLINE_FUNCTION void sort_thread(const TeamMember& t,
-                                        const ViewType& view,
-                                        const Comparator& comp) {
-  Impl::sort_nested_impl(t, view, nullptr, comp, Impl::NestedRange<false>());
-}
-
-template <class TeamMember, class KeyViewType, class ValueViewType>
-KOKKOS_INLINE_FUNCTION void sort_by_key_thread(const TeamMember& t,
-                                               const KeyViewType& keyView,
-                                               const ValueViewType& valueView) {
-  Impl::sort_nested_impl(t, keyView, valueView,
-                         Experimental::Impl::StdAlgoLessThanBinaryPredicate<
-                             typename KeyViewType::non_const_value_type>(),
-                         Impl::NestedRange<false>());
-}
+#ifndef KOKKOS_NESTED_SORT_HPP_
+#define KOKKOS_NESTED_SORT_HPP_
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_NESTED_SORT
+#endif
 
-template <class TeamMember, class KeyViewType, class ValueViewType,
-          class Comparator>
-KOKKOS_INLINE_FUNCTION void sort_by_key_thread(const TeamMember& t,
-                                               const KeyViewType& keyView,
-                                               const ValueViewType& valueView,
-                                               const Comparator& comp) {
-  Impl::sort_nested_impl(t, keyView, valueView, comp,
-                         Impl::NestedRange<false>());
-}
+#include "sorting/Kokkos_NestedSort.hpp"
```

Comment:
Can't this just be `sorting/Kokkos_NestedSort.hpp` avoiding the indirection?

Reply:
sorry don't understand, what do you mean?